bin/consequence_mapping/consequence_mapping.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+"""
+Pipeline for mapping variants to the genes they affect and their functional consequences, using Ensembl VEP API.
+For documentation, refer to /docs/consequence-mapping.md
+"""
+
+import itertools
+import json
+import logging
+import os
+import requests
+import sys
+
+from retry import retry
+
+logging.basicConfig()
+logger = logging.getLogger('consequence_mapping')
+logger.setLevel(logging.INFO)
+
+
+def deduplicate_list(lst):
+    """Removes duplicates from a list containing arbitrary (possibly unhashable) values."""
+    return [element for element, _ in itertools.groupby(sorted(lst))]
+
+
+def colon_to_vep(colon_id):
+    """Converts a colon-based identifier to VEP compatible one. Example: '15:7237571:C:T' → '15 7237571 . C T'"""
+    return '{} {} . {} {}'.format(*colon_id.split(':'))
+
+
+def vep_to_colon(vep_id):
+    """Converts a VEP compatible identifier to colon-based."""
+    vep_id_fields = vep_id.split(' ')
```

Comment:
Is there a strict specification for VEP IDs? Is the single white space requirement strictly adhered to in all the IDs?

Reply:
VEP support several types of variant IDs. This function converts only a single specific type of identifier (because we only query VEP with that one type of identifier). I expanded function docstring to explain that.